apps/omg_child_chain/test/omg_child_chain/block_queue/gas_analyzer_test.exs

```diff
@@ -0,0 +1,165 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.BlockQueue.GasAnalyzerTest do
+  use ExUnit.Case, async: true
+  import ExUnit.CaptureLog, only: [capture_log: 1]
+  alias OMG.ChildChain.BlockQueue.GasAnalyzer
+
+  setup do
+    {:ok, gas_analyzer} = GasAnalyzer.start_link(name: String.to_atom("test-#{:rand.uniform(1000)}"))
+    handler_id = {:gas_subbmission_handler, :rand.uniform(100)}
+
+    on_exit(fn ->
+      :telemetry.detach(handler_id)
+    end)
+
+    {:ok, %{gas_analyzer: gas_analyzer, handler_id: handler_id}}
+  end
+
+  describe "handle_info/2" do
+    test "queue gets emptied when gas gets callculated", %{
+      test: test_name,
+      gas_analyzer: gas_analyzer,
+      handler_id: handler_id
+    } do
+      defmodule test_name do
+        def eth_get_transaction_receipt(_), do: {:ok, %{"gasUsed" => "0x123"}}
+        def eth_get_transaction_by_hash(_), do: {:ok, %{"gasPrice" => "0x123"}}
+      end
+
+      # we're mocking ethereumex with our module
```

Comment:
I think this comment'd serve better in line 37 before `test_name` module definition.
Why not add a meaningful name to it then?